### ISSUE TITLE
AttributeError when pylint is run via python -m pylint from src directory

### ISSUE THREAD

**Issue by thisch** [PyCQA/pylint#2952]
### Steps to reproduce
1. Download pytest src
2. cd pytest/src/_pytest
3. python -m pylint logging.py  # raises AttributeError
4. pylint logging.py  # no AttributeError is raised

### Current behavior

If pylint is run using `python -m pylint` an AttributeError is output

```
> $ python3 -m pylint logging.py                                                               [±master ●●]
Could not import runpy module
Traceback (most recent call last):
  File "/home/thomas/miniconda/lib/python3.6/runpy.py", line 14, in <module>
    import importlib.machinery # importlib first so we can test #15386 via -m
  File "/home/thomas/miniconda/lib/python3.6/importlib/__init__.py", line 58, in <module>
    import warnings
  File "/home/thomas/gitrepos/pytest/src/_pytest/warnings.py", line 5, in <module>
    import pytest
  File "/home/thomas/miniconda/lib/python3.6/site-packages/pytest.py", line 8, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/thomas/miniconda/lib/python3.6/site-packages/_pytest/assertion/__init__.py", line 13, in <module>
    from _pytest.assertion import rewrite
  File "/home/thomas/miniconda/lib/python3.6/site-packages/_pytest/assertion/rewrite.py", line 9, in <module>
    import imp
  File "/home/thomas/miniconda/lib/python3.6/imp.py", line 31, in <module>
    warnings.warn("the imp module is deprecated in favour of importlib; "
AttributeError: module 'warnings' has no attribute 'warn'
```

### Expected behavior

No exception

### pylint --version output

```
> $ pylint --version                                                                           [±master ●●]
pylint 2.3.1
astroid 2.2.5
Python 3.6.7 | packaged by conda-forge | (default, Feb 28 2019, 09:07:38) 
[GCC 7.3.0]
```

**Comment by PCManticore** [comment]
Thanks for the report @thisch This is a third or fourth recent issue about the same thing, the fact that `pylint` appends the current working directory to the path, resulting in local modules shadowing builtin ones. It's something we need to address as soon as possible.

**Comment by thisch** [comment]
Here is a simple unittest that reproduces the problem.

```py
import subprocess
import sys


def test_issue2952(tmpdir):
    p_warnings = tmpdir / 'warnings.py'
    p_logging = tmpdir / 'logging.py'

    p_logging.write("")
    p_warnings.write("import mymodule")

    with tmpdir.as_cwd():
        # works
        output = subprocess.check_output(
            'pylint logging.py'.split()).decode()
        assert 'Traceback' not in output

        # fails
        output = subprocess.check_output(
            [sys.executable] + '-m pylint logging.py'.split()).decode()
        assert 'Traceback' not in output
```

**Comment by thisch** [comment]
I'm not sure if this can be fixed in pylint. The problem seems to be that

``python -m module-name``  seems to import ``runpy``, which imports warnings if run in the ``pytest/src/_pytest`` directory:

```
(pylint_venv_py37) > $ python -c "import runpy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.7/runpy.py", line 14, in <module>
    import importlib.machinery # importlib first so we can test #15386 via -m
  File "/usr/lib/python3.7/importlib/__init__.py", line 58, in <module>
    import warnings
  File "/home/thomas/gitrepos/pytest/src/_pytest/warnings.py", line 5, in <module>
    import pytest
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/pytest.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/_pytest/assertion/__init__.py", line 6, in <module>
    from _pytest.assertion import rewrite
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 16, in <module>
    import atomicwrites
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/atomicwrites/__init__.py", line 5, in <module>
    import tempfile
  File "/usr/lib/python3.7/tempfile.py", line 184, in <module>
    from random import Random as _Random
  File "/usr/lib/python3.7/random.py", line 40, in <module>
    from warnings import warn as _warn
ImportError: cannot import name 'warn' from 'warnings' (/home/thomas/gitrepos/pytest/src/_pytest/warnings.py)
```

I've read that ``python -m module-name args`` seems to be the same as ``python path-to-module/__main__.py args``, but this is not true, because runnint pylint using the latter command works in the ``pytes/tsrc/_pytest`` dir. 

Unfortunately it does not even work if python -m pylint is run from a different directory:

```
(pylint_venv_py37) > $ cd /tmp
(pylint_venv_py37) > $ python -m pylint ~/gitrepos/pytest/src/_pytest/logging.py                                             
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/thomas/gitrepos/pylint/pylint/__main__.py", line 13, in <module>
    pylint.run_pylint()
  File "/home/thomas/gitrepos/pylint/pylint/__init__.py", line 17, in run_pylint
    from pylint.lint import Run
  File "/home/thomas/gitrepos/pylint/pylint/lint.py", line 73, in <module>
    import astroid
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/__init__.py", line 63, in <module>
    from astroid.nodes import *
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/nodes.py", line 23, in <module>
    from astroid.node_classes import (
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/node_classes.py", line 38, in <module>
    from astroid import bases
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/bases.py", line 33, in <module>
    MANAGER = manager.AstroidManager()
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/util.py", line 26, in <lambda>
    lambda: importlib.import_module("." + module_name, "astroid")
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/manager.py", line 24, in <module>
    from astroid.interpreter._import import spec
  File "/home/thomas/pylint_venv_py37/lib/python3.7/site-packages/astroid/interpreter/_import/spec.py", line 12, in <module>
    import imp
  File "/usr/lib/python3.7/imp.py", line 31, in <module>
    warnings.warn("the imp module is deprecated in favour of importlib; "
AttributeError: module 'warnings' has no attribute 'warn'
```

**Comment by thisch** [comment]
Unfortunately this issue still persists:

```
(py38_pylint_master) > $ python -m pylint logging.py                                                                
Could not import runpy module
Traceback (most recent call last):
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/runpy.py", line 14, in <module>
    import importlib.machinery # importlib first so we can test #15386 via -m
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/importlib/__init__.py", line 58, in <module>
    import warnings
  File "/home/thomas/gitrepos/pytest/src/_pytest/warnings.py", line 6, in <module>
    import pytest
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/pytest/__init__.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/_pytest/assertion/__init__.py", line 9, in <module>
    from _pytest.assertion import rewrite
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 24, in <module>
    from _pytest.assertion import util
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/_pytest/assertion/util.py", line 14, in <module>
    import _pytest._code
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/_pytest/_code/__init__.py", line 2, in <module>
    from .code import Code  # noqa
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/_pytest/_code/code.py", line 27, in <module>
    import attr
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/attr/__init__.py", line 5, in <module>
    from . import converters, exceptions, filters, validators
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/attr/converters.py", line 7, in <module>
    from ._make import NOTHING, Factory
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/site-packages/attr/_make.py", line 7, in <module>
    import uuid
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/uuid.py", line 57, in <module>
    _AIX     = platform.system() == 'AIX'
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/platform.py", line 891, in system
    return uname().system
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/platform.py", line 857, in uname
    processor = _syscmd_uname('-p', '')
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/platform.py", line 611, in _syscmd_uname
    import subprocess
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/subprocess.py", line 684, in <module>
    class Popen(object):
  File "/home/thomas/miniconda/envs/py38_pylint_master/lib/python3.8/subprocess.py", line 935, in Popen
    def __del__(self, _maxsize=sys.maxsize, _warn=warnings.warn):
AttributeError: partially initialized module 'warnings' has no attribute 'warn' (most likely due to a circular import)
```

but 

```
(py38_pylint_master) > $ pylint logging.py                                                                          [±root_logger_refactoring ●]
************* Module logging
logging.py:267:0: C0301: Line too long (118/100) (line-too-long)
logging.py:288:0: C0301: Line too long (166/100) (line-too-long)
logging.py:292:0: C0301: Line too long (193/100) (line-too-long)
logging.py:328:0: C0301: Line too long (126/100) (line-too-long)
logging.py:356:0: C0301: Line too long (106/100) (line-too-long)
logging.py:394:0: C0301: Line too long (118/100) (line-too-long)
logging.py:396:0: C0301: Line too long (111/100) (line-too-long)
logging.py:399:0: C0301: Line too long (117/100) (line-too-long)
logging.py:418:0: C0301: Line too long (111/100) (line-too-long)
logging.py:422:0: C0301: Line too long (106/100) (line-too-long)
logging.py:425:0: C0301: Line too long (110/100) (line-too-long)
logging.py:444:0: C0301: Line too long (111/100) (line-too-long)
logging.py:458:0: C0301: Line too long (111/100) (line-too-long)
logging.py:462:0: C0301: Line too long (106/100) (line-too-long)
logging.py:778:0: C0301: Line too long (111/100) (line-too-long)
logging.py:781:0: C0301: Line too long (110/100) (line-too-long)
logging.py:2:0: W0406: Module import itself (import-self)
logging.py:30:28: E1101: Module 'logging' has no 'Formatter' member (no-member)
logging.py:36:8: E1101: Module 'logging' has no 'CRITICAL' member (no-member)
logging.py:37:8: E1101: Module 'logging' has no 'ERROR' member (no-member)
logging.py:38:8: E1101: Module 'logging' has no 'WARNING' member (no-member)
logging.py:39:8: E1101: Module 'logging' has no 'WARN' member (no-member)
logging.py:40:8: E1101: Module 'logging' has no 'INFO' member (no-member)
logging.py:41:8: E1101: Module 'logging' has no 'DEBUG' member (no-member)
logging.py:42:8: E1101: Module 'logging' has no 'NOTSET' member (no-member)
logging.py:59:29: E1101: Module 'logging' has no 'getLevelName' member (no-member)
logging.py:71:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:73:8: W0212: Access to a protected member _fmt of a client class (protected-access)
logging.py:30:0: R0903: Too few public methods (1/2) (too-few-public-methods)
logging.py:585:8: W0201: Attribute '_style' defined outside __init__ (attribute-defined-outside-init)
logging.py:77:28: E1101: Module 'logging' has no 'PercentStyle' member (no-member)
logging.py:125:8: R1705: Unnecessary "elif" after "return" (no-else-return)
logging.py:125:11: C0123: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
logging.py:127:13: C0123: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
logging.py:137:13: C0123: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
logging.py:143:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:77:0: R0903: Too few public methods (1/2) (too-few-public-methods)
logging.py:167:0: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:167:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
logging.py:180:51: W0622: Redefining built-in 'type' (redefined-builtin)
logging.py:274:18: E1101: Module 'logging' has no 'getLogger' member (no-member)
logging.py:294:29: E1101: Module 'logging' has no 'NOTSET' member (no-member)
logging.py:302:24: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:307:8: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:310:27: E1101: Module 'logging' has no 'LogRecord' member (no-member)
logging.py:313:8: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:315:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:317:8: W0201: Attribute 'stream' defined outside __init__ (attribute-defined-outside-init)
logging.py:341:21: E1101: Module 'logging' has no 'getLogger' member (no-member)
logging.py:351:45: E1101: Module 'logging' has no 'LogRecord' member (no-member)
logging.py:364:8: R1705: Unnecessary "else" after "return" (no-else-return)
logging.py:437:17: E1101: Module 'logging' has no 'getLogger' member (no-member)
logging.py:464:17: E1101: Module 'logging' has no 'getLogger' member (no-member)
logging.py:457:4: R0201: Method could be a function (no-self-use)
logging.py:487:4: W0212: Access to a protected member _finalize of a client class (protected-access)
logging.py:490:0: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:498:15: E1101: Module 'logging' has no 'WARNING' member (no-member)
logging.py:509:59: W0631: Using possibly undefined loop variable 'setting_name' (undefined-loop-variable)
logging.py:515:0: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:519:0: R0902: Too many instance attributes (11/7) (too-many-instance-attributes)
logging.py:533:12: C0415: Import outside toplevel (_pytest.warnings._issue_warning_captured) (import-outside-toplevel)
logging.py:534:12: C0415: Import outside toplevel (_pytest.deprecated.NO_PRINT_LOGS) (import-outside-toplevel)
logging.py:550:34: E1101: Module 'logging' has no 'Formatter' member (no-member)
logging.py:556:36: E1101: Module 'logging' has no 'FileHandler' member (no-member)
logging.py:565:33: W0108: Lambda may not be necessary (unnecessary-lambda)
logging.py:583:24: E1101: Module 'logging' has no 'Formatter' member (no-member)
logging.py:585:8: W0212: Access to a protected member _style of a client class (protected-access)
logging.py:586:12: W0212: Access to a protected member _fmt of a client class (protected-access)
logging.py:586:12: W0212: Access to a protected member _style of a client class (protected-access)
logging.py:630:32: E1101: Module 'logging' has no 'FileHandler' member (no-member)
logging.py:644:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:696:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:701:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:706:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:711:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:718:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:723:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:728:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:746:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:776:32: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:790:8: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:807:4: C0116: Missing function or method docstring (missing-function-docstring)
logging.py:824:12: E1101: Module 'logging' has no 'StreamHandler' member (no-member)
logging.py:816:16: W0201: Attribute '_first_record_emitted' defined outside __init__ (attribute-defined-outside-init)
logging.py:823:16: W0201: Attribute '_section_name_shown' defined outside __init__ (attribute-defined-outside-init)

------------------------------------------------------------------
Your code has been rated at 4.38/10 (previous run: 4.38/10, +0.00)
```

works.


```
> pylint --version
pylint 2.6.0-dev1
astroid 2.5.0
Python 3.8.2 | packaged by conda-forge | (default, Mar 23 2020, 18:16:37) 
[GCC 7.3.0]

**Comment by PCManticore** [comment]
So if you check the first traceback, you can see the import doesn't even reach pylint at all and it's caused by runpy and importlib. I am on my phone right and can't really check for you, but there is an issue in Python 3.8+ in their bug tracker which specifies the fault problem with `-m` and runpy. The linked changes are the most we can do for pylint and if I recall correctly, other tools like flake8 failed to run as well in this situation.

**Comment by thisch** [comment]
Thx, didn't know that this issue could be due to a bug in runpy. Probably you mean https://bugs.python.org/issue29206.

**Comment by PCManticore** [comment]
@thisch I'm referring to https://bugs.python.org/issue33053. It seems to have been fixed in 3.8, but as we can see, that's not the case.
